src/sentry/static/sentry/app/views/organizationEventsV2/tags.jsx

```diff
@@ -1,18 +1,88 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import {isEqual} from 'lodash';
 
 import SentryTypes from 'app/sentryTypes';
 import TagDistributionMeter from 'app/components/tagDistributionMeter';
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import {fetchTags} from './utils';
 
-export default class Tags extends React.Component {
+class Tags extends React.Component {
   static propTypes = {
+    api: PropTypes.object.isRequired,
+    organization: SentryTypes.Organization.isRequired,
     view: SentryTypes.EventView.isRequired,
+    selection: SentryTypes.GlobalSelection,
   };
 
+  state = {
+    isLoading: true,
+    tags: {},
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.view.id !== prevProps.view.id ||
+      !isEqual(this.props.selection, prevProps.selection)
+    ) {
+      this.fetchData();
+    }
+  }
+
+  fetchData() {
+    const {api, organization, view} = this.props;
+    fetchTags(api, organization.slug, view.tags)
+      .then(tags => {
+        this.setState({tags, isLoading: false});
+      })
+      .catch(() => {
+        this.setState({tags: {}, isLoading: false});
+      });
+  }
+
   renderTag(tag) {
-    return <TagDistributionMeter key={tag} title={tag} segments={[]} />;
+    const {isLoading, tags} = this.state;
+    let segments = [];
+    let totalValues = 0;
+    if (!isLoading && tags[tag] && tags[tag]) {
+      totalValues = tags[tag].totalValues;
+      segments = tags[tag].topValues;
+    }
+
+    segments.forEach(segment => {
+      segment.url = 'TODO';
```

Comment:
I have to cook up a function to generate URLs in #13470 perhaps we could re-use that same logic here?

Reply:
Updated to use this function, I changed the signature slightly since we don't have a tag object of the same structure here